### PR TITLE
general SEO improvements for all pages.

### PR DESCRIPTION
**Pull request by god-hephaestus**
Finally closing #58 after a long delay. I'm hoping these work. Here are some notes in case we want to revisit this topic later:

Main page seems good to me so I haven't done any changes. But some SEO tools rate it 3/100 or something. If we want to improve I suggest a complete rewrite.

Planes, Tools, Homebrews, Library(?), Languages(?), Tables and Races(?) are untouched for now. Question marks are for tabs that already had some mentions of Teothe in it.

Generators don't have any text before fetch so I couldn't improve it. Maybe some long SEO text explaining the generators and mentioning Teothe could be nice.

Lets see how these changes play out and we'll go from there.

### CODE REVIEW

**Review by ozgurozbek — CHANGES_REQUESTED** [review]
I realize the changes you've done, and fairly happy with them. I also understand your decisions. Thank you.

I also have some dislikes about the PR, I'll hold this before merging as I don't want to occupy crawlers and wait for another month before pushing. I'll merge after the checklist below is complete.

- [ ] Please refer to the review for line reviews.
- [ ] Could you also adjust the meta description and SEO? Pages will need to generate them individually. Any URL from the web-app still use the same title and meta description. Current one doesn't even fit mobile display. See example:
image 
- [x] Title now has our keyword since you've changed it to Teothe but since every title is the same our slugs might cause critical failures. IIRC I've paid attention to them when I was first doing them because SEO was a concern anyway. See slugs: https://www.semrush.com/blog/what-is-a-url-slug/ I'll tick this for the time being, you might want to take a look.
- [ ] Could you also add more direct links to the website in general, for example adding a button or a hyperlink for directing people to deities or planes, in the common lore page? This will help crawlers to demonstrate our cornerstones. See https://yoast.com/what-is-cornerstone-content/#utm_source=yoast-seo&utm_medium=software&utm_term=learn-about-cornerstone&utm_content=content-analysis

**Comment by god-hephaestus** [comment]
Requested and necessary changes are done. Every page now has a component and the page function has the component in it for content. This way we can use metadata while having hooks on every page.

Only conflict is the navbar, which groups things differently with the new DM's Works tab.

File: src/app/layout.tsx
Comment (ozgurozbek) [inline src/app/layout.tsx:12]: Great usage of templates as discussed, thank you.

File: src/app/update-calendar/page.tsx
Comment (ozgurozbek) [inline src/app/update-calendar/page.tsx:236]: What is going on here? Linting or indentation changes? DIFF is weird but works and okay.

File: src/app/sitemap.xml
Comment (ozgurozbek) [inline src/app/sitemap.xml:1]: I forgot about those, thanks.

File: src/app/tools/page.tsx
Comment (ozgurozbek) [inline src/app/tools/page.tsx:7]: I mean, not exactly ours but fair work 👯

**Review by ozgurozbek — APPROVED** [review]

**Comment by ozgurozbek** [comment]
> Every page now has a component and the page function has the component in it for content.

This is how it was when we were using Pages instead of App Router. Should've kept it that way. Now our SSR should be improved too, since it's all best-practice.

_Great work miner, rock and stone._

**Pull request merged**
